src/redux/modules/mdb.js

```diff
@@ -71,9 +71,14 @@ const stripOldFiles = unit => {
     }
 
     return acc.concat(val);
-  }, []).map(f => f.duration ? f : { ...f, duration: unit.duration });
+  }, []).map(f => {
+    if (!f.duration) {
+      f.duration = unit.duration;
+    }
+    return f;
+  });
 
-  return { ...unit, files: nFiles };
+  return unit;
```

Comment:
You don't have to return unit. Or even you should not as you both "change" it and "return" it...
Immer will not know which one to use .... (this is probably not a bug as both are the same...) but probably not best practice....

https://redux-toolkit.js.org/usage/immer-reducers#mutating-and-returning-state

Reply:
Now I'm not sure as it looks like a helper function and not the "state update function"...

Reply:
I think it is okay.

Reply:
this is helper function, and its return value is used, so I'd not change it